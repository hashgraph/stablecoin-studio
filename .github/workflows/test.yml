name: test
on:
  pull_request:
    branches: [ main, release/**, feature/** ]
  push:
    branches: [ main, release/** ]
    tags: [ v* ]
jobs:
  test-node:
    name: Unit Tests
    runs-on: ubuntu-latest
    steps:
      - name: Use Node.js TLS 16.x
        uses: actions/setup-node@v1
        with:
          node-version: 16.x
      - name: Checkout repository
        uses: actions/checkout@v2
      - name: Install dependencies
        run: npm i
      - name: Build
        run: npm run build
      - name: Test
        run: npm run test