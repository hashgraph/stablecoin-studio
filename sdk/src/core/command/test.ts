/* eslint-disable @typescript-eslint/no-empty-interface */

interface CommandResponse {}
interface ICommand {}

export class Command<T> implements ICommand {
	private $resultType!: T;
}

export interface ICommandBus {
	execute<X>(command: Command<X>): Promise<X>;
}

export type IInferringCommandHandler<CommandType extends Command<unknown>> =
	CommandType extends Command<infer ResultType>
		? {
				execute(command: CommandType): Promise<ResultType>;
		  }
		: never;

class ConcreteCommandResponse implements CommandResponse {
	constructor(public readonly status: string) {}
}

class ConcreteCommand extends Command<ConcreteCommandResponse> {
	constructor(public readonly value: string) {
		super();
	}
}

class CommandHandler<CommandType extends Command<unknown>>
	implements IInferringCommandHandler<ConcreteCommand>
{
	execute(command: ConcreteCommand): Promise<ConcreteCommandResponse> {
		throw new Error('Method not implemented.');
	}
}

class CommandBus implements ICommandBus {
	execute<X>(command: Command<X>): Promise<X> {
		throw new Error('Method not implemented.');
	}
}

// Test

const bus = new CommandBus();
const res = bus.execute(new ConcreteCommand('1'));
