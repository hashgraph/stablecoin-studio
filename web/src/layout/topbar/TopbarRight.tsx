import { Box, Flex, HStack, Image, Text, VStack } from '@chakra-ui/react';
import { useDispatch, useSelector } from 'react-redux';
import Icon from '../../components/Icon';
import SDKService from '../../services/SDKService';
import { SELECTED_WALLET_PAIRED, walletActions } from '../../store/slices/walletSlice';
import type { SavedPairingData } from 'hedera-stable-coin-sdk';
import HEDERA_LOGO from '../../assets/png/hashpackLogo.png';
import METAMASK_LOGO from '../../assets/svg/MetaMask_Fox.svg';
import TooltipCopy from '../../components/TooltipCopy';

const TopbarRight = () => {
	const dispatch = useDispatch();

	const pairingData: SavedPairingData = useSelector(SELECTED_WALLET_PAIRED);
	const dAppName = pairingData?.metadata?.name;

	const handleDisconnect = () => {
		SDKService.getInstance().then((instance) => instance?.disconectHaspack());

		dispatch(walletActions.clearData());
		dispatch(walletActions.setSelectedStableCoin(undefined));
		dispatch(walletActions.setStableCoinList([]));
	};

	const getIcon = (): string => {
		if (dAppName === 'HashPack') return HEDERA_LOGO;

		return METAMASK_LOGO;
	};

	return (
		<Flex data-testid='topbar-right' gap={5}>
			<HStack
				color='white'
				borderRadius='8px'
				h='46px'
				minW='150px'
				justifyContent={'space-between'}
				px={3}
				boxShadow='2px 2px 10px 0px #f1f1f1, -2px -2px 10px 0px #FFF'
				bgColor='#4a4a4a'
			>
				<VStack spacing={0}>
					<TooltipCopy valueToCopy={pairingData ? pairingData.accountIds[0] : ''}>
						<Text data-testid='topbar-right-account' fontSize='12px' fontWeight={600}>
							{pairingData ? pairingData.accountIds[0] : ''}
						</Text>
					</TooltipCopy>
					<Text data-testid='topbar-right-network' fontSize='10px' textTransform='uppercase'>
						{pairingData ? pairingData.network : ''}
					</Text>
				</VStack>
				<Image src={getIcon()} alt={dAppName} w='25px' h='25px' alignSelf='center' />
			</HStack>
			<Box borderLeft='2px solid' borderLeftColor='light.primary' w='1px' />
			<Flex
				onClick={handleDisconnect}
				h='32px'
				minW='32px'
				borderRadius='50%'
				bgColor='light.purple4'
				justifyContent='center'
				alignItems='center'
				alignSelf='center'
				_hover={{
					cursor: 'pointer',
					bgColor: 'light.purple2',
				}}
			>
				<Icon name='Power' fontSize='24px' color='dark.primary' />
			</Flex>
		</Flex>
	);
};

export default TopbarRight;
