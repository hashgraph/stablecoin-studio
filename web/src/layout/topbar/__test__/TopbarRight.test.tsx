import TopbarRight from '../TopbarRight';
import { render } from '../../../test/';
import configureMockStore from 'redux-mock-store';

const networkName = 'TESNET';
const accountId = '0.0.123';
const mockStore = configureMockStore();
const store = mockStore({
	wallet: {
		data: {
			savedPairings: [{ network: networkName, accountIds: [accountId] }],
			topic: '',
			pairingString: '',
			encryptionKey: '',
		},
	},
});
describe(`<${TopbarRight.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<TopbarRight />, store);

		const topbar = component.getByTestId('topbar-right');
		expect(topbar).toBeInTheDocument();
		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should show network', () => {
		const component = render(<TopbarRight />, store);

		const network = component.getByTestId('topbar-right-network');
		expect(network).toBeInTheDocument();
		expect(network).toHaveTextContent(networkName);
	});

	test('should show operating account', () => {
		const component = render(<TopbarRight />, store);

		const account = component.getByTestId('topbar-right-account');
		expect(account).toBeInTheDocument();
		expect(account).toHaveTextContent(accountId);
	});
});
