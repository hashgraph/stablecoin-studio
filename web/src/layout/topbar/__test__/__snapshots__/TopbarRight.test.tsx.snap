// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopbarRight /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-166re2f"
    data-testid="topbar-right"
  >
    <div
      class="chakra-stack css-d17m5h"
    >
      <div
        class="chakra-stack css-jp20jv"
      >
        <div
          class="css-1m8iww1"
        >
          <p
            class="chakra-text css-o1au3i"
            data-testid="topbar-right-account"
          >
            0.0.123
          </p>
        </div>
        <p
          class="chakra-text css-gegx7f"
          data-testid="topbar-right-network"
        >
          TESNET
        </p>
      </div>
      <img
        class="chakra-image css-1lzenv"
        src="MetaMask_Fox.svg"
      />
    </div>
    <div
      class="css-dv2lk7"
    />
    <div
      class="css-1a9f5vk"
    >
      <svg
        class="chakra-icon css-dee54c"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 256 256"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <line
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
          x1="128"
          x2="128"
          y1="48"
          y2="124"
        />
        <path
          d="M176,54.2a88,88,0,1,1-96,0"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
      </svg>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
